src/SFA.DAS.AANHub.Api.UnitTests/Controllers/EmployersControllerTests.cs

```diff
@@ -71,5 +71,101 @@ public EmployersControllerTests()
 
             result.As<BadRequestObjectResult>().StatusCode.Should().Be(StatusCodes.Status400BadRequest);
         }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetEmployer_InvokesQueryHandler(
+            ValidatedResponse<GetEmployerMemberResult> handlerResult)
+        {
+            _mediator.Setup(m => m.Send(It.IsAny<GetEmployerMemberQuery>(), It.IsAny<CancellationToken>())).ReturnsAsync(handlerResult);
+
+            var userRef = new Guid();
+            var response = await _controller.GetEmployer(userRef);
+            response.Should().NotBeNull();
+
+            var result = response as OkObjectResult;
+            Assert.AreEqual(StatusCodes.Status200OK, result!.StatusCode);
+
+            var queryResult = result.Value as GetEmployerMemberResult;
+            queryResult.Should().BeEquivalentTo(handlerResult.Result);
+
+            _mediator.Verify(m => m.Send(It.IsAny<GetEmployerMemberQuery>(), It.IsAny<CancellationToken>()));
+        }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetEmployer_InvokesQueryHandler_NoResultGivesNotFound()
+        {
+            var errorResponse = new ValidatedResponse<GetEmployerMemberResult>
+                (new List<ValidationFailure>());
+
+            var userRef = Guid.NewGuid();
+            _mediator.Setup(m => m.Send(It.Is<GetEmployerMemberQuery>(q => q.UserRef == userRef),
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(errorResponse);
+
+            var response = await _controller.GetEmployer(userRef);
+
+            var result = response as NotFoundResult;
+            result.Should().NotBeNull();
+            Assert.AreEqual(StatusCodes.Status404NotFound, result!.StatusCode);
+
+            _mediator.Verify(m => m.Send(It.IsAny<GetEmployerMemberQuery>(), It.IsAny<CancellationToken>()));
+        }
+
+        [Test]
+        [AutoMoqData]
+        public async Task GetEmployer_InvokesQueryHandler_ResultGivesSuccessfulResult()
+        {
+            var response = new ValidatedResponse<GetEmployerMemberResult>
+            (new GetEmployerMemberResult
+            {
+                Email = "email@email.com",
+                MemberId = new Guid(),
+                Name = "name"
+            });
+
+            var userRef = Guid.NewGuid();
+            _mediator.Setup(m => m.Send(It.Is<GetEmployerMemberQuery>(q => q.UserRef == userRef),
+                    It.IsAny<CancellationToken>()))
+                .ReturnsAsync(response);
+
+            var getResult = await _controller.GetEmployer(userRef);
+
+            var result = getResult as OkObjectResult;
+            result.Should().NotBeNull();
+            Assert.AreEqual(StatusCodes.Status200OK, result!.StatusCode);
```

Comment:
Redundant verification, status code 200 is implicit with OkObjectResult. I would rather verify the result.Value has expected outcome.

Reply:
Test now compares result, and mocked response.